### DISCUSSION TITLE
Config migration adds a potentially redundant "{/,}" suffix when migrating matchPackagePrefixes to matchPackageNames

### DISCUSSION THREAD

**Discussion by Egorand** [renovatebot/renovate#33218] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### Please tell us more about your question or problem

Our project uses Renovate to manage Gradle packages, and most artifacts are declared using the "group:artifact:version" notation. Our renovate.json5 config file contained a bunch of package rules that used `matchPackagePrefixes` to match packages. Renovate recently opened a PR migrating our config to use `matchPackageNames` instead (as it seems that `matchPackagePrefixes` was removed), and migrations looked roughly like this:

```diff
- matchPackagePrefixes: [
+ matchPackageNames: [
-   'de.undercouch:gradle-download-task',
+   'de.undercouch:gradle-download-task{/,}**',
]
```

If my understanding is correct, the `{/,}` portion of the pattern matches an optional `/` character. However, Gradle package names don't usually include the `/` character (in fact, I'm not even sure if it's an allowed character), therefore that suffix is redundant in our case. Is it possible that this suffix (introduced in https://github.com/renovatebot/renovate/pull/30394) targets a specific type of packages? 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by secustor** [comment]
Without `{/,}` the rule would for example not match `de.undercouch:gradle-download-task-foo-bar`

**Reply by Egorand** [reply]
Interesting, can you please elaborate? 

I might be wrong, but wouldn't `'de.undercouch:gradle-download-task**'` be enough to match both `de.undercouch:gradle-download-task` and `de.undercouch:gradle-download-task-foo-bar`, whereas `'de.undercouch:gradle-download-task{/,}**'` would also match `de.undercouch:gradle-download-task/foo-bar`? The latter is not a valid Gradle/Maven package coordinate, therefore `{/,}` is redundant.

**Reply by rarkins** [reply]
matchPackageNames is not specific to maven. This pattern is the one which is backwards compatible. You are free to change it once migrated or prior to migration

**Reply by Egorand** [reply]
Thanks, I understand that package name matching is not supposed to be specific to a particular package type, and that I can tweak it after the migration. My original question was basically this: isn't the `{/,}` matcher trying to match a specific type of packages (that include the `/` character), and wouldn't `**` achieve the same effect without being unnecessarily package type-specific?

**Reply by secustor** [reply]
Docker images with registries prefixed, could have for example multiple slashes.

**Reply by Egorand** [reply]
Presumably there's a Docker image named `de.undercouch:gradle-download-task//docker-image` (apologies if it's an invalid name, I'm not familiar with Docker image naming rules), wouldn't `'de.undercouch:gradle-download-task**'` match it?

**Reply by rarkins** [reply]
You can use online glob testers to test your theory

https://www.digitalocean.com/community/tools/glob

foo** does not match foo/bar

**Reply by Egorand** [reply]
On the other hand, `'de.undercouch:gradle-download-task{/,}**'` does not match `de.undercouch:gradle-download-task-bar/foo`, whereas previously I'd have expected a match when using `'de.undercouch:gradle-download-task'` as a package prefix matcher.

Anyway, it seems that `{/,}` is present in the migration to try to match one or more slashes that immediately follow the package prefix, and it presumably targets specific package types that allow slashes in their names. I'm all for the extra safety this potentially provides, and having to perform minor manual cleanup is no big deal.
